plugins/module_utils/module_container/docker_api.py

```diff
@@ -746,10 +746,11 @@ def _preprocess_etc_hosts(module, client, api_version, value):
 def _preprocess_healthcheck(module, client, api_version, value):
     if value is None:
         return value
-    if not value or not value.get('test'):
+    if not value or not (value.get('test') or (value.get('test_cli_compatible') and value.get('test') is None)):
         value = {'test': ['NONE']}
-    elif 'test' in value:
+    elif value.get('test'):
```

Comment:
For `test_cli_compatible=false` this should stay `'test' in value`. So maybe:
```suggestion
    elif (value.get('test') if value.get('test_cli_compatible') else 'test' in value):
```
What's the reason to change it for `test_cli_compatible=true`?

Reply:
No reason I believed there was the test key with None value but it's not.